dreadnode/main.py

```diff
@@ -163,6 +179,121 @@ def _get_profile_api_key(self, profile: str | None = None) -> str | None:
         # Silently fail if profile config is not available or invalid
         return None
 
+    def _resolve_organization(self) -> None:
+        if self._api is None:
+            return
+
+        if self.organization:
+            self._organization = self._api.get_organization(self.organization)
+            if not self._organization:
+                raise RuntimeError(f"Organization '{self.organization}' not found.")
+
+        else:
+            organizations = self._api.list_organizations()
+
+            if not organizations:
+                raise RuntimeError(
+                    f"You are not part of any organizations on {self.server}. You will not be able to use Strikes.",
+                )
+
+            if len(organizations) > 1:
+                # We should not presume to choose an organization
+                org_list = "\t\n".join([f"- {o.name}" for o in organizations])
+                raise RuntimeError(
+                    f"You are part of multiple organizations. Please specify an organization from:\n{org_list}"
+                )
+            self._organization = organizations[0]
+
+    def _create_workspace(self, name: str) -> Workspace:
+        if self._api is None:
+            raise RuntimeError("API client is not initialized.")
+
+        try:
+            return self._api.create_workspace(name=name, organization_id=self._organization.id)
+        except RuntimeError as e:
+            if "403: Forbidden" in str(e):
+                raise RuntimeError(
+                    "You do not have permission to create the specified workspace for this organization. (You must be a Contributor to create new workspaces)."
+                ) from e
+            raise
+
+    def _resolve_workspace(self) -> None:
+        if self._api is None:
+            return
+
+        found_workspace: Workspace | None = None
+        if self.workspace:
+            try:
+                found_workspace = self._api.get_workspace(self.workspace)
+            except RuntimeError as e:
+                if "404: Workspace not found" in str(e):
+                    pass  # do nothing, we'll create it below
```

Comment:
Consider adding an exception when user doesn't have access to the workspace or returns 403 from the api `access denied to the wrokspace`

```python
if "404" ...
elif "403" in str(e):
        # User can't access this workspace - don't create, error ot
        raise RuntimeError(
            f"Access denied to workspace '{self.workspace}'. "
            "You don't have permission to access this workspace."
        ) from e
```